controller/destination/server.go

```diff
@@ -104,6 +127,32 @@ func (s *server) Get(dest *common.Destination, stream pb.Destination_GetServer)
 	return nil
 }
 
+func (s *server) localKubernetesServiceIdFromDNSName(host string) (*string, error) {
+	hostLabels := strings.Split(host, ".")
+
+	// Verify that `host` ends with .svc.$zone.
+	if len(hostLabels) <= 1 + len(s.k8sDNSZoneLabels) {
+		return nil, nil
```

Comment:
second return value here should be a non-nil error, I think.  And likewise for all the returns in this method.